Add more integer overflow tests for `enclosingIntRect` in FloatRectTests.cpp

Ahmad-S792 opened the pull request:
#### e73b8072d7b44b4246dd5a7dfdbad3de3b085d38
<pre>
Add more integer overflow tests for `enclosingIntRect` in FloatRectTests.cpp

<a href="https://bugs.webkit.org/show_bug.cgi?id=267162">https://bugs.webkit.org/show_bug.cgi?id=267162</a>

Reviewed by Antti Koivisto.

Partial Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/663b023cf7022eca119acc1b8b3a610cd8993da3">https://chromium.googlesource.com/chromium/src.git/+/663b023cf7022eca119acc1b8b3a610cd8993da3</a>

This PR adds additional test from Blink commit and the underlying bug was already fixed in 189573@main.

* Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
(EnclosingIntRect):

Canonical link: <a href="https://commits.webkit.org/272779@main">https://commits.webkit.org/272779@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/947fbc4a34eac0a81090c38fc1aad5046726bd9f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33004 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11776 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34905 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35647 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29824 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33974 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8945 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29257 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33478 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9924 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29489 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8652 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8806 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29445 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36981 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29981 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29857 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34936 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8943 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6884 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32791 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10634 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7661 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9534 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9573 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2644ea8c6d482446ccec1f142aec20f755f1b518. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22493

aprotyas commented on Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
```diff
@@ -722,6 +723,39 @@ TEST(FloatRect, EnclosingIntRect)
+    EXPECT_EQ(WebCore::IntRect(-101, -151, 201, 351), WebCore::enclosingIntRect(fractionalNegposRect3));
+
+    WebCore::FloatRect maxRect(-std::numeric_limits<float>::max() / 2, -std::numeric_limits<float>::max() / 2, std::numeric_limits<float>::max(), std::numeric_limits<float>::max());
+    EXPECT_EQ(WebCore::IntRect(INT_MIN, INT_MIN, INT_MAX, INT_MAX), WebCore::enclosingIntRect(maxRect));
```
Nit: I'd prefer if we just used the `constexpr` min/max functions instead of the macros.

```suggestion
    constexpr int intMin = std::numeric_limits<int>::min();
    constexpr int intMax = std::numeric_limits<int>::max();
    EXPECT_EQ(WebCore::IntRect(intMin, intMin, intMax, intMax), WebCore::enclosingIntRect(maxRect));
```

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/947fbc4a34eac0a81090c38fc1aad5046726bd9f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/22493

Ahmad-S792 committed:
Add more integer overflow tests for `enclosingIntRect` in FloatRectTests.cpp

https://bugs.webkit.org/show_bug.cgi?id=267162

Reviewed by Antti Koivisto.

Partial Merge: https://chromium.googlesource.com/chromium/src.git/+/663b023cf7022eca119acc1b8b3a610cd8993da3

This PR adds additional test from Blink commit and the underlying bug was already fixed in 189573@main.

* Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
(EnclosingIntRect):

Canonical link: https://commits.webkit.org/272779@main

webkit-commit-queue commented:
Committed 272779@main (e73b8072d7b4): <https://commits.webkit.org/272779@main>

Reviewed commits have been landed. Closing PR #22493 and removing active labels.
